lib/src/model_utils.dart

```diff
@@ -166,3 +169,150 @@ String crossdartifySource(
   }
   return newSource;
 }
+
+/// An UnmodifiableListView that computes equality and hashCode based on the
+/// equality and hashCode of its contained objects.
+class HashableList extends UnmodifiableListView<dynamic> {
+  HashableList(Iterable<dynamic> iterable) : super(iterable);
+
+  @override
+  bool operator ==(other) {
```

Comment:
As written, this is not symmetric, since HashableList will compare equal to an ordinary list, but not vice versa.  Do you need this behavior?

If not, I'd suggest adding an if test, like so:
```dart
@override
bool operator ==(other) {
  if (other is HashableList) {
    if (this.length == other.length) {
      ...
    }
  }
  return false;
}
```

As a side benefit, the operations on `other` will no longer require dynamic dispatches, so they will be faster on some implementations.